scalalib/src/mill/scalalib/Assembly.scala

```diff
@@ -192,11 +192,11 @@ object Assembly {
     ).pathRef
   }
 
-  def create(
+  private[mill] def create0(
       destJar: os.Path,
       inputPaths: Agg[os.Path],
       manifest: mill.api.JarManifest = mill.api.JarManifest.MillDefault,
-      prependShellScript: Option[String] = None,
+      prepend: Option[os.Source] = None,
```

Comment:
I assume you need a byte array instead of a string. Woud be a `Option[String|Array[Byte]]` sufficient? Or do you expect large data here? A `os.Source` feels a bit too generic.

Reply:
`Option[String|Array[Byte]]` would work as well. The data is around 300K in this case.

Reply:
Basic question: What is the simplest way to pass a `x: String | Array[Byte]` to `os.write`? Looks like there is no implicit conversion for unions. `os.write(..., x)` does not work. `x: os.Source`, `x.asInstanceOf[os.Source]` doesn't work either. Is there any other way that's less clunky than `x match { case x: String => os.write(..., x); case x: Array[Byte] => os.write(..., x) }`?

Reply:
I don't know the answer. Could be, this isn't fully implemented in upickle. But we could also either fallback to always use a byte array, which sounds a bit inconvenient for the other cases, or go with a `Option[Either[Array[Byte],String]]` for now.

Reply:
I think just using byte array is fine. Strings can be converted with just `.getBytes`, so it's no big hardship. Definitely easier than wrapping in `Either`s and `Left`s and `Right`s